zero_bin/rpc/src/native/txn.rs

```diff
@@ -74,14 +75,28 @@ where
         gas_used: tx_receipt.gas_used as u64,
     };
 
-    let (code_db, tx_traces) = match (pre_trace, diff_trace) {
+    let (code_db, mut tx_traces) = match (pre_trace, diff_trace) {
         (
             GethTrace::PreStateTracer(PreStateFrame::Default(read)),
             GethTrace::PreStateTracer(PreStateFrame::Diff(diff)),
         ) => process_tx_traces(access_list, read, diff).await?,
         _ => unreachable!(),
     };
 
+    // Handle case when transaction failed and a contract creation was reverted
+    if !tx_status && tx_receipt.contract_address.is_some() {
+        tx_traces.insert(
+            tx_receipt.contract_address.unwrap(),
+            TxnTrace {
+                balance: None,
+                nonce: None,
+                storage_read: None,
+                storage_written: None,
+                code_usage: None,
+            },
```

Comment:
Would be cleaner to just derive `Default` for `TxnTrace`

Reply:
Right, seems nothing problematic with deriving `Default` on `TxnTrace` type, currently not present.